# **Exercise 1**

In [2]:
while True:
    print('*', end='')

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

# **Exercise 2**

In [2]:
print = 'PRINT'

In [3]:
print('Hello World')

TypeError: 'str' object is not callable

# **Exercise 3**

In [25]:
squares = [i**2 for i in range(10)]
print("List comprehension:", squares_comp)


List comprehension: [0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [28]:
squares_loop = list()
for i in range(10):
    squares_loop.append(i**2)
print("For loop:", squares_loop)

For loop: [0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [6]:
%%timeit  
squares = [i**2 for i in range(10)]

361 ns ± 3.07 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


It took 361 nanoseconds for the code to run. 
The standard deviation (variation) is 3.07 nanoseconds.
The actual time could be 3.07ns more or less than 362ns.

In [30]:
%%timeit 
squares_loop = list()
for i in range(10):
    squares_loop.append(i**2)

388 ns ± 6.71 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


The code took 388 nanoseconds to run. The variation for running this code is 6.71 nanoseconds. The actual time taken to run the code could be 6.71ns more or less than 388ns.